### PR TITLE
docs: ✏️ Securing a node with SSL/TLS

### PR DESCRIPTION
**Pull request by heldrida**

### CODE REVIEW

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+
+Learn more about record types [here](https://en.wikipedia.org/wiki/List_of_DNS_record_types).
+
+DNS changes may take some time to propagate, thus make sure to [verify the DNS records](#how-to-verify-dns-records).
+
+### Alternatively, getting a free domain name for testing?
+
+Domain names are costly, you might not have one, or not be interested at this time! For this reason, we're going to provide some hints on how to set up a free domain name for your server.
+
+There are plenty of free domain server providers but for this guide will opt for [Ngrok](https://ngrok.com), there are other alternatives such as [Localhost.run](localhost.run) or [LocalTunnel.me](https://localtunnel.me), [Ngrok](https://ngrok.com/), but we're keeping things simple here and will not try to provide support for anything else, but should be quite similar!
```
Comment (ozwaldorf) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md]:
```suggestion
There are plenty of free domain server providers, but for this guide we will opt for [Ngrok](https://ngrok.com). There are other alternatives such as [Localhost.run](localhost.run) or [LocalTunnel.me](https://localtunnel.me), but we're keeping things simple here and will not try to provide support for anything else. Using an alternative should be quite similar! 
```

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+dig fleek-network-node.fleek.xyz +nostats +nocomments +nocmd
+```
+
+```sh
+;fleek-network-node.fleek.xyz.		IN	A
+fleek-network-node.fleek.xyz.	390	IN	A	dyna.wikimedia.org.
+```
+
+👆 The goal is to determine if the DNS record matches the machine IP Address we set up earlier. There might be cases you'll need to wait longer, but it's good to double-check your DNS record dashboard and revise it if the desired settings fail to show!
+
```
Comment (ozwaldorf) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md:204]: hmm doesn't seem to have the ip address in the example response

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+```
+
+For our demo, we replace `ursa.earth` to `fleek-network-node.fleek.xyz`
+
+```sh
+# SSL code
+ssl_certificate /etc/letsencrypt/live/fleek-network-node.fleek.xyz/fullchain.pem;
+ssl_certificate_key /etc/letsencrypt/live/fleek-network-node.fleek.xyz/privkey.pem;
+```
+
```
Comment (ozwaldorf) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md:323]:
```suggestion

> Note: if multiple domain names are used, the `init-letsencrypt.sh` script will export them under the first domain provided

```

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 4]
```diff
@@ -0,0 +1,527 @@
...
+The directory file structure should look like:
+
+```sh
+.
+├── README.md
+├── data
+├── docker-compose.yml
+└── init-letsencrypt.sh
+```
```
Comment (ozwaldorf) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md:382]: Maybe this bit of info could be better put farther above where the chmod +x is done

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 5]
```diff
@@ -0,0 +1,527 @@
...
+Since we have already:
+
+- Updated the `A Records` in the [previous step](#how-to-set-up-the-dns-settings-for-a-node-server)
+- [Verified](#how-to-verify-dns-records) that the DNS records are correct
+
+We can execute the `init-letsencrypt.sh` by prefixing the script with the variables `EMAIL` and `DOMAINS`.
+
+- DOMAINS - A list separated by space, e.g. `DOMAINS="foobar.com my-node.fleek.xyz"`
+- EMAIL - A valid email address that the server administrator, the ops operator or anyone of concern should have access to, e.g. `ops@fleek.xyz`
+
```
Comment (ozwaldorf) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md:399]: for redundancy:
```suggestion

> Note: if multiple domains are provided, the certificates will be stored under the first one

```

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 6]
```diff
@@ -0,0 +1,527 @@
...
+```sh
+Successfully received certificate.
+Certificate is saved at: /etc/letsencrypt/live/fleek-network-node.fleek.xyz/fullchain.pem
+Key is saved at:         /etc/letsencrypt/live/fleek-network-node.fleek.xyz/privkey.pem
+This certificate expires on 2023-04-25.
+These files will be updated when the certificate renews.
+```
+
+You may also find that the paragraph after, that mentions that the certificate expires is important. If you are running the Docker Stack we provide, the `certbot` is installed and automating the certification for you.
```
Comment (ozwaldorf) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md:479]:
```suggestion
You may also find that the paragraph after, that mentions that the certificate expires is important. If you are running the Docker Stack we provide, the `certbot` is installed and automates renewing the certification for you.
```

**Review by ozwaldorf — CHANGES_REQUESTED** [review]

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 7]
```diff
@@ -0,0 +1,527 @@
...
+
+In recent years, the non-profit organization [Let's Encrypt](https://en.wikipedia.org/wiki/Let's_Encrypt) started providing TLS certificate encryption, that is free of charge. The Let's Encrypt organization also improved the administration experience by providing automation features for system administrator processes.
+
+In that respect, we'll look into how to secure a Fleek Node with TLS certificates provided by the nonprofit Certificate Authority (CA) Let's Encrypt, by harnessing the knowledge provided in our [Running a node in a Docker container](fleek-network-running-a-node-in-a-docker-container) guide. We'll look into how to set up our custom domain in the Nginx configuration, generate dummy certificates, and real certificates and understand how the certificate automation is handled to mitigate issues, such as expiration.
+
+The knowledge demonstrated here should be useful and can be easily applied to native setups (none Docker). As some troubleshooting and differences are found in different systems, we'll stick with our Docker Stack for that matter and simplicity!
```
Comment (theBeardA) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md:48]:
```suggestion
The knowledge demonstrated here should be useful and can be easily applied to native setups (not Docker). As some troubleshooting and differences are found in different systems, we'll stick with our Docker Stack for that matter and simplicity!
```

the `none` doesn't sound right here 🤔

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 8]
```diff
@@ -0,0 +1,527 @@
...
+| Type        | Host                              | Answer            | TTL         |
+| ----------- | --------------------------------- | ----------------- | ----------- |
+| A           | fleek-network-node.fleek.xyz      |  181.196.118.156  | 300         |
+
+
+Learn more about record types [here](https://en.wikipedia.org/wiki/List_of_DNS_record_types).
+
+DNS changes may take some time to propagate, thus make sure to [verify the DNS records](#how-to-verify-dns-records).
+
+### Alternatively, getting a free domain name for testing?
```
Comment (nicolaspoggi) [inline guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md:119]:
```suggestion
### Alternatively, how can one get a free domain name for testing?
```

**Review by nicolaspoggi — CHANGES_REQUESTED** [review]
Solid write! Can't validate technically fully - but the guide is clear, left a minor suggestion, and then one more global suggestion is reducing the use of "!" to end sentences which in this case stood out.

File: guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md [hunk 9]
```diff
@@ -0,0 +1,527 @@
...
+![](./assets/fleek-network-how-to-secure-a-network-node.png?202301251236)
+
+## Introduction
+
+Secure Sockets Layer or Transport Layer Security are the technologies used to keep an internet connection secure by protecting sensitive data that is transferred between systems. It's a protocol for servers and web clients to ensure that data passed between them is private!
+
+A secure connection is required to prevent others from modifying, reading or tempering the data that is transmitted. Otherwise, it can be intercepted by third parties, such as hackers.
+HTTP connections are insecure by default, SSL or TLS (the modern version of SSL) is used to encrypt the data by using advanced encryption algorithms in transit, preventing others from reading it as it's transferred over the connection.
+
+💡 Generally referred to as SSL, or TLS (Transport Layer Security), the modern version that's usually named SSL)
```
[the remark this answers is not in the document]
Comment (heldrida) [reply guides/Network nodes/fleek-network-securing-a-node-with-ssl-tls.md]: Done

**Review by ozwaldorf — APPROVED** [review]

**Pull request merged**